internal/bminventory/inventory.go

```diff
@@ -250,11 +250,27 @@ func (b *bareMetalInventory) updatePullSecret(pullSecret string, log logrus.Fiel
 func (b *bareMetalInventory) GetDiscoveryIgnition(ctx context.Context, params installer.GetDiscoveryIgnitionParams) middleware.Responder {
 	log := logutil.FromContext(ctx, b.log)
 
+	cluster, err := common.GetClusterFromDB(b.db, params.ClusterID, common.SkipEagerLoading)
+	if err != nil {
+		log.WithError(err).Errorf("failed to get cluster: %s", params.ClusterID)
+		return common.GenerateErrorResponder(err)
+	}
+
 	infraEnv, err := common.GetInfraEnvFromDB(b.db, params.ClusterID)
 	if err != nil {
+		log.WithError(err).Errorf("failed to get infra-env: %s", params.ClusterID)
 		return common.GenerateErrorResponder(err)
 	}
 
+	// update pull secret and proxy, since they could have been set during cluster update
+	proxy := models.Proxy{
+		HTTPProxy:  swag.String(cluster.HTTPProxy),
+		HTTPSProxy: swag.String(cluster.HTTPSProxy),
+		NoProxy:    swag.String(cluster.NoProxy),
+	}
+	infraEnv.PullSecret = cluster.PullSecret
+	infraEnv.Proxy = &proxy
```

Comment:
Do we also need to handle ignition config override and static networking here?
Presumably these could also have been updated since the cluster was created.

Reply:
I guess static networking is less of a concern because it's only set in the generate endpoint and it looks like we're updating the infra env there, but I think this still applies for the override because that's a separate endpoint.

Reply:
Good catch. Fixed the ignition config overide. Static Networking is only passed during Generate Cluster ISO call, and infra env is updated with it